File: watcher-frontend/src/types/metricas.ts
export interface MetricasGenerales {
  // Presupuestarias
  total_programas: number;
  monto_total_inicial: number;
  monto_total_vigente: number;
  monto_total_ejecutado: number;
  porcentaje_ejecucion_global: number;
  
  // Actos
  total_actos: number;
  actos_alto_riesgo: number;
  actos_medio_riesgo: number;
  actos_bajo_riesgo: number;
  
  // Alertas
  total_alertas: number;
  alertas_criticas: number;
  alertas_altas: number;
  
  // Vinculación
  total_vinculos: number;
  tasa_vinculacion: number;
  
  // Top 5
  top_organismos_presupuesto: Array<{
    organismo: string;
    monto: number;
  }>;
  top_organismos_riesgo: Array<{
    organismo: string;
    actos_alto_riesgo: number;
  }>;
}

export interface MetricasOrganismo {
  organismo: string;
  total_programas: number;
  monto_inicial: number;
  monto_vigente: number;
  monto_ejecutado: number;
  porcentaje_ejecucion: number;
  total_actos: number;
  actos_alto_riesgo: number;
  total_alertas: number;
  alertas_criticas: number;
}

export interface DistribucionRiesgo {
  por_nivel: Record<string, number>;
  por_tipo_acto: Record<string, Record<string, number>>;
  por_organismo: Record<string, Record<string, number>>;
  monto_por_nivel: Record<string, number>;
}

File: watcher-backend/app/schemas/metricas.py
"""
Schemas for Métricas endpoints
"""

from typing import Any

from pydantic import BaseModel


class MetricasGeneralesResponse(BaseModel):
    # Presupuestarias
    total_programas: int
    monto_total_inicial: float
    monto_total_vigente: float
    monto_total_ejecutado: float
    porcentaje_ejecucion_global: float

    # Actos
    total_actos: int
    actos_alto_riesgo: int
    actos_medio_riesgo: int
    actos_bajo_riesgo: int

    # Alertas
    total_alertas: int
    alertas_criticas: int
    alertas_altas: int

    # Vinculación
    total_vinculos: int
    tasa_vinculacion: float

    # Top 5
    top_organismos_presupuesto: list[dict[str, Any]]
    top_organismos_riesgo: list[dict[str, Any]]

class MetricasOrganismoResponse(BaseModel):
    organismo: str
    total_programas: int
    monto_inicial: float
    monto_vigente: float
    monto_ejecutado: float
    porcentaje_ejecucion: float
    total_actos: int
    actos_alto_riesgo: int
    total_alertas: int
    alertas_criticas: int

class DistribucionRiesgoResponse(BaseModel):
    por_nivel: dict[str, int]
    por_tipo_acto: dict[str, dict[str, int]]
    por_organismo: dict[str, dict[str, int]]
    monto_por_nivel: dict[str, float]

